BossMod/Modules/Stormblood/Dungeon/D03BardamsMettle/D032HunterOfBardam.cs

```diff
@@ -0,0 +1,183 @@
+namespace BossMod.Stormblood.Dungeon.D03BardamsMettle.D032HunterOfBardam;
+
+public enum OID : uint
+{
+    Boss = 0x1AA5, // R2.200, x1
+    Bardam = 0x1AA3, // R15.000, x1
+    ThrowingSpear = 0x1F49, // R1.25
+    StarShard = 0x1F4A, // R2.4
+    LoomingShadow = 0x1F4D, // R1.0
+    WarriorOfBardam = 0x1AA4, // R1.1
+    Helper = 0x18D6
+}
+
+public enum AID : uint
+{
+    Visual1 = 9450, // Helper->player, no cast, single-target
+    Visual2 = 9451, // Helper->player, no cast, single-target
+    Visual3 = 9611, // StarShard->self, no cast, single-target
+
+    Magnetism = 7944, // Boss->self, no cast, range 40+R circle, pull 40 between hitboxes
+    Tremblor1 = 9596, // Helper->self, 4.0s cast, range 10 circle
+    Tremblor2 = 9595, // Helper->self, 4.0s cast, range 10-20 donut
+    EmptyGaze = 7940, // Boss->self, 6.5s cast, range 40+R circle
+    Travail = 7935, // Bardam->self, no cast, single-target
+    Charge = 9599, // ThrowingSpear->self, 2.5s cast, range 45+R width 5 rect
+
+    Sacrifice = 7937, // Helper->location, 7.0s cast, range 3 circle, tower
+    DivinePunishment = 7938, // Helper->self, no cast, range 40+R circle
+
+    BardamsRing = 9601, // Helper->self, no cast, range 10-20 donut, stack donuts, 5.3s delay
+    CometFirst = 9597, // Helper->location, 4.0s cast, range 4 circle
+    CometRest = 9598, // Helper->location, 1.5s cast, range 4 circle
+
+    HeavyStrike = 9591, // Boss/WarriorOfBardam->self, 4.0s cast, single-target
+    HeavyStrike1 = 9592, // Helper->self, 4.0s cast, range 6+R 270-degree cone
+    HeavyStrike2 = 9593, // Helper->self, 4.0s cast, range 6+R-12+R 270-degree donut segment
+    HeavyStrike3 = 9594, // Helper->self, 4.0s cast, range 12+R-18+R 270-degree donut segment
+
+    CometImpact = 9600, // StarShard->self, 4.0s cast, range 9 circle
+    ReconstructVisual = 7933, // Bardam->self, no cast, single-target
+    Reconstruct = 7934, // Helper->location, 4.0s cast, range 5 circle
+
+    Tremblor = 9605, // Boss->self, 3.5s cast, single-target
+    MeteorImpact = 9602, // LoomingShadow->self, 30.0s cast, ???
+}
+
+public enum IconID : uint
+{
+    BardamsRing = 58, // player
+    ChasingAOE = 197, // player
+}
+
+class Comet(BossModule module) : Components.LocationTargetedAOEs(module, ActionID.MakeSpell(AID.CometFirst), 4);
+class CometFirst(BossModule module) : Components.LocationTargetedAOEs(module, ActionID.MakeSpell(AID.CometFirst), 4);
```

Comment:
why do we have two components for CometFirst?

Reply:
seems like he copied it incorrectly, on BMR it is 
class Comet(BossModule module) : Components.StandardChasingAOEs(module, new AOEShapeCircle(4), ActionID.MakeSpell(AID.CometFirst), ActionID.MakeSpell(AID.CometRest), 10, 1.5f, 9, true, (uint)IconID.ChasingAOE);
class CometFirst(BossModule module) : Components.LocationTargetedAOEs(module, ActionID.MakeSpell(AID.CometFirst), 4);
class CometRest(BossModule module) : Components.LocationTargetedAOEs(module, ActionID.MakeSpell(AID.CometRest), 4);

but on vbm the AI for chasingAOEs is broken anyway, it will often get stuck inside AOEs (this affects both old and new pathfinding algorithm) because it cant decide where to go. i made sort of a hack to circumvent the problem

    public override void AddAIHints(int slot, Actor actor, PartyRolesConfig.Assignment assignment, AIHints hints)
    {
        if (Chasers.Count > 0)
        {
            AddForbiddenZones(actor, hints, true);
            AddForbiddenZones(actor, hints, false);
        }
    }

    private void AddForbiddenZones(Actor actor, AIHints hints, bool isTarget)
    {
        // sort of a hack to prevent the AI from getting "stuck" inside the AOE because all paths to safety have equal distance
        foreach (var c in Chasers.Where(c => c.Target == actor == isTarget))
        {
            var circle = (AOEShapeCircle)c.Shape;
            var radius = isTarget ? MoveDistance + circle.Radius : circle.Radius + 1;
            var position = isTarget ? c.PredictedPosition() - circle.Radius * actor.Rotation.ToDirection() : c.PredictedPosition();
            hints.AddForbiddenZone(ShapeDistance.Circle(position, radius), c.NextActivation);
        }
    }

Reply:
yeah my version of bardam's doesn't even use chasing AOEs, they have a long enough cast time that ai can always easily avoid them

Reply:
Yes I modified it last minute to remove the chasing aorncuae that shit didn't work and as you said there's plenty of time to avoid this way. Can r3move the duplicate.

Reply:
well BMR got many components modified and framework changes, so just copying the modules alone will not always give the intended results.
i know it can technically be dodged by the cast time alone, 1.5s is plenty for AI when the circle is only 5 diameter (needs about 833ms to run out). But i make my modules not just for a single use case in mind so I added it for non AI users since they dont have 1 frame reaction times like AI.